File: packages/strategy-service/internal/asset/workspace/template/js_basic/backtest/engine.js
// 回测引擎占位，只输出当前策略骨架收集到的模块结果。
function run(ctx, sizing) {
  return {
    bars: ctx.bars.length,
    factors: ctx.values.length,
    filter: ctx.filter.length,
    entry: ctx.entry.length,
    exit: ctx.exit.length,
    sizing: sizing(0),
    ready: false,
  }
}

module.exports = { run }

File: packages/strategy-service/internal/asset/workspace/template/python_basic/backtest/engine.py
def run(ctx, sizing):
    """回测引擎占位，只输出当前策略骨架收集到的模块结果。"""

    return {
        "bars": len(ctx["bars"]),
        "factors": len(ctx["values"]),
        "filter": len(ctx["filter"]),
        "entry": len(ctx["entry"]),
        "exit": len(ctx["exit"]),
        "sizing": sizing(0),
        "ready": False,
    }
